gammapy/maps/tests/test_hpxmap.py

```diff
@@ -382,3 +382,26 @@ def test_plot_poly():
     m = HpxNDMap.create(binsz=10)
     with mpl_plot_check():
         m.plot(method="poly")
+
+def test_hpxmap_addition_subtraction():
+    map1 = HpxMap.create(nside=8, unit='cm2')
+    map2 = HpxMap.create(nside=8, unit='m2')
+
+    map1 += 1*u.cm**2
```

Comment:
You mostly have tests with `+=` and other in-place operators, that will call `__iadd__`.
Usually in-place operators are used for in-place modifications, but in your initial implementation, always a copy was made.
So I find this a bit misleading / hard to understand.

If you change to the implementation that I suggested, where `__add__` make the copy and `__iadd__` is in-place, then there should be at least one explicit example for each case, for `__add__` with an assert showing that the initial map data isn't modified.